### ISSUE TITLE
[CLI] `ImportError: cannot import name '_get_mock_module' from 'pytest_mock'`

### ISSUE THREAD

**Issue by samuela** [wandb/client#2682]
**Description**
The pytest test suite fails with 
```
ImportError while loading conftest '/build/source/tests/conftest.py'.
tests/conftest.py:15: in <module>
    from tests import utils
tests/utils/__init__.py:7: in <module>
    from tests.utils.mock_backend import BackendMock
tests/utils/mock_backend.py:8: in <module>
    from pytest_mock import _get_mock_module  # type: ignore
E   ImportError: cannot import name '_get_mock_module' from 'pytest_mock' (/nix/store/cn86h8xap96v0xxfzypg5zr8xjbzccgm-python3.9-pytest-mock-3.6.1/lib/python3.9/site-packages/pytest_mock/__init__.py)
```
when using pytest-mock v3.6.1.

**Wandb features**
n/a

**How to reproduce**
Build this nix derivation:
<details>
  <summary>source</summary>

```nix
{ buildPythonPackage
, flask
, pytest-mock
, click
, pytestCheckHook
, configparser
, docker_pycreds
, fetchFromGitHub
, matplotlib
, bokeh
, git
, GitPython
, lib
, pathtools
, promise
, protobuf
, psutil
, python
, python-dateutil
, pyyaml
, requests
, sentry-sdk
, shortuuid
, subprocess32 ? null
, yaspin
}:

buildPythonPackage rec {
  pname = "wandb";
  version = "0.12.2";

  src = fetchFromGitHub {
    owner = pname;
    repo = "client";
    rev = "v${version}";
    sha256 = "0rdh7vgb3njj9xi8kjfm4735zw3slzvc0qsvsc7nrl65d8iwn0j3";
  };

  # The wandb requirements.txt does not distinguish python2/3 dependencies. We
  # need to drop the subprocess32 dependency when building for python3.
  patchPhase = lib.optional python.isPy3 ''
    sed -i 's/^subprocess32>=.*//' requirements.txt
  '';

  # git is not a setup.py dependency of wandb, but wandb does expect git to be
  # in PATH.
  propagatedBuildInputs = [
    click
    configparser
    docker_pycreds
    git
    GitPython
    pathtools
    promise
    protobuf
    psutil
    python-dateutil
    pyyaml
    requests
    sentry-sdk
    shortuuid
    yaspin
  ] ++ lib.optionals python.isPy2 [ subprocess32 ];

  # The wandb tests require network access, and so cannot be run in the nix
  # build environment.
  # doCheck = false;
  checkInputs = [
    pytestCheckHook
    bokeh
    flask
    pytest-mock
    matplotlib
  ];

  pythonImportsCheck = [ "wandb" ];

  meta = with lib; {
    description = "A CLI and library for interacting with the Weights and Biases API";
    homepage = "https://github.com/wandb/client";
    license = licenses.mit;
    maintainers = with maintainers; [ samuela ];
  };
}
```
</details>

**Environment**
- OS: NixOS
- Environment: n/a
- Python Version: 3.9

**Comment by vanpelt** [comment]
Hey @samuela our dev environment currently requires [pytest-mock<=3.2.0](https://github.com/wandb/client/blob/master/tox.ini#L14).  Are you able to install 3.2.0?

**Comment by samuela** [comment]
Unfortunately no, at least not easily. Nixpkgs tries to avoid supporting outdated releases of packages unless absolutely necessary.

**Comment by vanpelt** [comment]
@samuela I just cut a new branch that should handle the newer pytest-mock versions.  You can try it now with:

```
pip install --upgrade git+git://github.com/wandb/client.git@task/pytest-mock#egg=wandb
```

It should go out in the official release next week.

**Comment by samuela** [comment]
Unfortunately I'm still seeing the same error as of 0.12.3 (latest at the time of writing). Here's the full nix derivation: https://gist.github.com/samuela/f7aa10ba81c0571c74e45284c9d3ee19. Error is
```
ImportError while loading conftest '/build/source/tests/conftest.py'.
tests/conftest.py:15: in <module>
    from tests import utils
tests/utils/__init__.py:7: in <module>
    from tests.utils.mock_backend import BackendMock
tests/utils/mock_backend.py:8: in <module>
    from pytest_mock import _get_mock_module  # type: ignore
E   ImportError: cannot import name '_get_mock_module' from 'pytest_mock' (/nix/store/44qsxih9gavdfvjvpc7kiad68m1sm1hi-python3.9-pytest-mock-3.6.1/lib/python3.9/site-packages/pytest_mock/__init__.py)
```

**Comment by vanpelt** [comment]
@samuela apologies, this didn't make it into the last release.  It will make it into the next one.  Until then you can install from the branch as mentioned above.

**Comment by samuela** [comment]
I just tested this on 0.12.4, and it's still broken with the same error.

**Comment by samuela** [comment]
This seems to be fixed as of 0.12.7. Thanks!
